canarytokens/tokens.py

```diff
@@ -234,26 +236,26 @@ def look_for_source_data(query_name: str) -> Dict[str, str]:
     #     data["linux_inotify_filename_access"] = base64.b32decode(filename)
     #     return data
 
-    # @staticmethod
-    # def _generic(matches: Match[AnyStr]) -> Dict[str, str]:
-    #     data = {}
-    #     generic_data = matches.group(1)
-    #     generic_data = generic_data.replace(".", "").upper()
-    #     # this channel doesn't have padding, add if needed
-    #     # TODO: put this padding logic into utils somewhere.
-    #     generic_data += "=" * int(
-    #         (math.ceil(float(len(generic_data)) / 8) * 8 - len(generic_data)),
-    #     )
-    #     try:
-    #         # TODO: this can smuggle in all sorts of data we need to sanitise
-    #         #
-    #         data["generic_data"] = base64.b32decode(generic_data)
-    #     except TypeError:
-    #         data["generic_data"] = f"Unrecoverable data: {generic_data}"
-    #     return data
+    @staticmethod
+    def _generic(matches: Match[AnyStr]) -> dict[str, str]:
+        data = {}
+        generic_data = matches.group(1)
+        generic_data = generic_data.replace(".", "").upper()
+        # this channel doesn't have padding, add if needed
+        # TODO: put this padding logic into utils somewhere.
+        generic_data += "=" * int(
```

Comment:
Consider: using `//` (https://docs.python.org/3/library/operator.html#operator.__floordiv__) and `.ljust`